<launch>
    <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
        <!-- Remap topic required by the node to custom topics -->
        <remap from="image_rect" to="/balrog_camera/image_raw" />
        <remap from="camera_info" to="/balrog_camera/camera_info" />

        <!-- Optional: Subscribe to the compressed stream-->
        <param name="image_transport" type="str" value="compressed" />

        <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
        <param name="tag_family" type="str" value="36h11" />

        <!-- Describe the tags -->
        <rosparam param="tag_descriptions">[{id: 0, size: 0.215, frame_id: tag_0},
                                            {id: 1, size: 0.215, frame_id: tag_1},
                                            {id: 2, size: 0.215, frame_id: tag_2},
                                            {id: 3, size: 0.215, frame_id: tag_3},
                                            {id: 4, size: 0.215, frame_id: tag_4},
                                            {id: 5, size: 0.215, frame_id: tag_5},
                                            {id: 6, size: 0.215, frame_id: tag_6},
                                            {id: 7, size: 0.215, frame_id: tag_7},
                                            {id: 8, size: 0.215, frame_id: tag_8}]
        </rosparam>
    </node>

    <!-- publish a static transform from /base_footprint (center of Balrog) to /balrog_camera (the RPI camera), this is needed to be able to lookup the transform to apriltags -->
    <node pkg="tf" type="static_transform_publisher" name="camera_static_tf" args="0.0 0.0 .25 0.5 -0.5 0.5 -0.5 /base_footprint /balrog_camera 100"/>
</launch>
